src/main/kotlin/no/nav/syfo/sykmelding/service/EgenmeldtSykmeldingService.kt

```diff
@@ -56,16 +65,27 @@ class EgenmeldtSykmeldingService(private val oppdaterTopicsService: OppdaterTopi
         }
         database.registrerEgenmeldtSykmelding(egenmeldtSykmelding)
 
-        oppdaterTopicsService.oppdaterOKTopic(opprettReceivedSykmelding(fnr = fnr, sykmeldingId = egenmeldtSykmelding.id.toString()))
+        oppdaterTopicsService.oppdaterOKTopic(opprettReceivedSykmelding(egenmeldtSykmelding))
     }
 
-    fun opprettReceivedSykmelding(fnr: String, sykmeldingId: String): ReceivedSykmelding {
+    suspend fun opprettReceivedSykmelding(egenmeldtSykmelding: EgenmeldtSykmelding): ReceivedSykmelding {
         val fellesformatMarshaller: Marshaller = JAXBContext.newInstance(XMLEIFellesformat::class.java, XMLMsgHead::class.java, HelseOpplysningerArbeidsuforhet::class.java).createMarshaller()
             .apply { setProperty(Marshaller.JAXB_ENCODING, "UTF-8") }
 
+        val fnr = egenmeldtSykmelding.fnr
+        val sykmeldingId = egenmeldtSykmelding.id.toString()
+
+        val aktoerIds = aktoerIdClient.getAktoerIds(listOf(fnr), sykmeldingId)
+        val patientIdents = aktoerIds[fnr]
+
+        if (patientIdents == null || patientIdents.feilmelding != null) {
+            throw AktoerNotFoundException("Patient with fnr: " + fnr + " not found in registry, error: " + (patientIdents?.feilmelding
```

Comment:
Her kan du bruke kotlin sin nydelige string interpolation :D Altså, "Patient with fnr: $fnr not found..." En av de små forbedringene med kotlin i forhold til java som jeg er skikkelig glad i, selv om det kanskje bare er en detalj ;) 

I tillegg til å kaste feilen bør du også logge den som error. Det er ikke alltid det logges selv om vi kaster error, og det gjør feilsøking vanskelig. Logg gjerne sykmeldingsiden det feilet for også, så blir det enkelt å spore kallet :) 

Jeg synes forøvrig loggmeldingene bør være på norsk ;)

Reply:
Og så tenker jeg dette bør flyttes til klienten, som jeg skrev i en annen kommentar, så slipper den kallende klassen å forholde seg til "the inner workings" for aktør-domenet ;)